packages/frontend/src/scripts/upload.ts

```diff
@@ -30,6 +30,32 @@ const mimeTypeMap = {
 	'image/png': 'png',
 } as const;
 
+// tar.gzなど、拡張子内にドットを2つまで許容するものはここに追加
+const specialExtensions = [
+	'gz',
+	'bz2',
+	'xz',
+	'zst',
+	'lz',
+	'lz4',
+	'sz',
+	'z',
+	'zstd',
```

Comment:
二重拡張子になるファイルがありそうなパターンを追加で調査/検討してみました。
いかがでしょうか?
```suggestion
	'zstd',
	'zip',
	'7z',
	'gpg',
	'bak',
	'old',
	'diff',
	'patch',
```